From: Martin Konrad <konrad@frib.msu.edu>
Date: Mon, 11 Jan 2016 10:50:56 -0500
Subject: Fix MMIO under VxWorks with C++

This is patch is listed on the R3.15.3 "Know Problems" page.
---
 src/libCom/osi/os/vxWorks/epicsMMIO.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/libCom/osi/os/vxWorks/epicsMMIO.h b/src/libCom/osi/os/vxWorks/epicsMMIO.h
index 06301fd..dd8abe9 100644
--- a/src/libCom/osi/os/vxWorks/epicsMMIO.h
+++ b/src/libCom/osi/os/vxWorks/epicsMMIO.h
@@ -83,13 +83,18 @@
  * They do *not* all implement the sys{In/Out}{Byte/Word/Long}
  * functions to do the same thing though, so we can't use them.
  */
-
+#ifdef __cplusplus
+extern "C" {
+#endif
 UINT8 sysPciInByte(UINT8 *addr);
 void sysPciOutByte(UINT8 *addr, UINT8 data);
 UINT16 sysPciInWord(UINT16 *addr);
 void sysPciOutWord(UINT16 *addr, UINT16 data);
 UINT32 sysPciInLong (UINT32 *addr);
 void sysPciOutLong (UINT32 *addr, UINT32 data);
+#ifdef __cplusplus
+}
+#endif
 
 #define ioread8(address)           sysPciInByte((UINT8 *)(address))
 #define iowrite8(address,data)     sysPciOutByte((UINT8 *)(address), (epicsUInt8)(data))
